src/activity_handlers.cpp

```diff
@@ -695,6 +707,15 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
             on_ground = source_stack.begin();
             std::advance( on_ground, act.values.at( 1 ) );
             liquid = *on_ground;
+            break;
+        case LST_MONSTER:
+            Creature *c = g->critter_at(source_pos);
+            source_mon = dynamic_cast<monster *>(c);
+            if (source_mon == nullptr) {
+                throw std::runtime_error("could not find source creature for liquid transfer");
```

Comment:
Last time I checked, we don't use `throw` for non-critical things. Just `debugmsg` and try to restore to some semi-sane state (just cancel everything).